### DISCUSSION TITLE
What files to look at strictly for the bot config, upload (including vector store), and chat functionality

### DISCUSSION THREAD

**Discussion by d3buggerdan** [run-llama/chat-llamaindex#106] [Q&A]
I tested out the chat-llamaindex demo with my own documents and am pleased with the responses.

I am looking to strictly get the backend code that handles 
1. Configuring a openai api with system and assistant prompts
2. Uploading a document then creating the embeddings in a local vector store 
3. The chat interactions between a user and the bot that accesses the relevant document embeddings

I tried to dig into the code but am a little confused exactly what to extract (since there is a lot of additional code) that will essentially allow me to replicate this but on a simple level without all of the extra functionality such as UI/Settings etc. 

I essentially would like to have a simple project where I can have a folder to place a pdf, i can then run a script to take this and similar to the upload feature create embeddings and store them locally, and inside the terminal be able to chat with the document and receive the same answers as I would in the demo without all of the extra code.

If anyone can provide some clarity or direction to what I need to specifically look at would be greatly appreciated. Thanks.

**Comment by marcusschiesser** [comment]
Chat LlamaIndex is a bit more complicated as it supports multiple bots and indexes. If you're looking for a more simplified example, I recommend starting with https://github.com/run-llama/create-llama. It generates personalized example code with just one bot and one index. Chat LlamaIndex is based on it, too.
